File: core/src/com/quillraven/game/core/ecs/system/RemoveSystem.java
package com.quillraven.game.core.ecs.system;

import com.badlogic.ashley.core.Entity;
import com.badlogic.ashley.core.Family;
import com.badlogic.ashley.systems.IteratingSystem;
import com.quillraven.game.core.ecs.component.RemoveComponent;

public class RemoveSystem extends IteratingSystem {
    public RemoveSystem() {
        super(Family.all(RemoveComponent.class).get());
    }

    @Override
    protected void processEntity(final Entity entity, final float deltaTime) {
        this.getEngine().removeEntity(entity);
    }
}

File: core/src/main/kotlin/com/github/quillraven/mysticgarden/system/RemoveSystem.kt
package com.github.quillraven.mysticgarden.system

import com.github.quillraven.fleks.Entity
import com.github.quillraven.fleks.IteratingSystem
import com.github.quillraven.fleks.World.Companion.family
import com.github.quillraven.mysticgarden.component.Remove

class RemoveSystem : IteratingSystem(family { all(Remove) }) {

    override fun onTickEntity(entity: Entity) {
        with(entity[Remove]) {
            if (delay <= 0f) {
                entity.remove()
                return
            }

            delay -= deltaTime
        }
    }
}